streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -36,13 +37,10 @@ case class BatchInfo(
     streamIdToInputInfo: Map[Int, StreamInputInfo],
     submissionTime: Long,
     processingStartTime: Option[Long],
-    processingEndTime: Option[Long]
+    processingEndTime: Option[Long],
+    numOutputOp: Int
```

Comment:
I think instead of the `numOutputOp`, we need to give references to the OutputOperationInfo. Kind of link SparkListenerJobStart event has StageInfos in it.